.ruff.toml

```diff
@@ -12,20 +12,45 @@ select = [
   "E",
   "F",
   "W",
-  #"UP",
-  #"PT"
+  "N",
```

Comment:
I would really be interested in maybe another PR that just enables all of the rules and we see what happens to ndcube.

Reply:
I can do this for you, there'll be a catastrophic amount of changes to be made

Reply:
I don't mean for you to fix it. 

Just enable all the rules, autofix what can be fixed, open that PR and then do nothing else. Don't fix any fails yourself. I want to see what can be autofixed, what can't be and what needs to be fixed manually.

Reply:
So I opened the said PR and ran the unsafe-fixes on it, have a lookse